scripts/release/create_chart_release_pr.py

```diff
@@ -0,0 +1,122 @@
+import argparse
+import os
+import shutil
+import subprocess
+import sys
+import tempfile
+from pathlib import Path
+
+from github import Github, GithubException
+
+from lib.base_logger import logger
+
+REPO_URL = "https://github.com/mongodb/helm-charts.git"
+REPO_NAME = "mongodb/helm-charts"
+TARGET_CHART_SUBDIR = "charts/mongodb-kubernetes"
+BASE_BRANCH = "main"
+
+
+# run_command runs the command `command` from dir cwd
+def run_command(command, cwd=None):
+    logger.info(f"Running command: {' '.join(command)} in directory {cwd}")
+    result = subprocess.run(command, capture_output=True, text=True, cwd=cwd)
+    if result.returncode != 0:
+        raise RuntimeError(f"Command {' '.join(command)} failed. Stdout: {result.stdout}, stderr: {result.stderr}")
+    logger.info("Command succeeded")
+    return result.stdout
+
+
+# create_pull_request creates the pull request to the helm-charts repo
+def create_pull_request(branch_name, chart_version, github_token):
+    logger.info("Creating the pull request in the helm-charts repo.")
+
+    try:
+        g = Github(github_token)
+        repo = g.get_repo(REPO_NAME)
+        pr_title = f"Release MCK {chart_version}"
+        body = f"This PR publishes the MCK chart version {chart_version}."
+
+        pr = repo.create_pull(
+            title=pr_title,
+            body=body,
+            head=branch_name,
+            base=BASE_BRANCH,
+        )
+        logger.info(f"Successfully created Pull Request {pr.html_url}")
+    except Exception as e:
+        pr_url = f"https://github.com/{REPO_NAME}/pull/new/{branch_name}"
+        raise Exception(
+            f"An unexpected error occurred while creating the PR: {e}. Please create the PR manually by following this link {pr_url}"
+        )
+
+
+def commit_and_push_chart(chart_version):
+    branch_name = f"mck-release-{chart_version}"
+
+    mck_dir = Path(".").resolve()
+    # source_chart_path is local helm chart in MCK repo
+    source_chart_path = os.path.join(mck_dir, "helm_chart")
+
+    if not os.path.isdir(source_chart_path):
+        raise Exception(f"The source chart path '{source_chart_path}' is not a valid directory.")
+
+    github_token = os.environ.get("GH_TOKEN")
+    if not github_token:
+        raise Exception("github token not found. Returning because git push will fail.")
+
+    with tempfile.TemporaryDirectory() as temp_dir:
+        helm_repo_path = os.path.join(temp_dir, "helm-charts")
+        logger.info(f"Working in a temporary directory: {temp_dir}")
```

Comment:
nit: I also think that most logging here is debug level

`logger.info(f"Successfully created Pull Request {pr.html_url}")`: this is IMO important and deserves to be at info level, most other things are details and small steps. It's important to log them but debug is enough